pkg/controller/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -628,29 +633,33 @@ func withConditions(conditions ...toolchainv1alpha1.Condition) nsTmplSetOption {
 }
 
 func newNamespace(tier, username, typeName string, options ...namespaceOption) *corev1.Namespace {
+	labels := map[string]string{
+		"toolchain.dev.openshift.com/owner":    username,
+		"toolchain.dev.openshift.com/type":     typeName,
+		"toolchain.dev.openshift.com/provider": "codeready-toolchain",
+	}
+	if tier != "" {
+		labels["toolchain.dev.openshift.com/templateref"] = NewTierTemplateName(tier, typeName, "abcde11")
+		labels["toolchain.dev.openshift.com/tier"] = tier
+	}
 	ns := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: fmt.Sprintf("%s-%s", username, typeName),
-			Labels: map[string]string{
-				"toolchain.dev.openshift.com/tier":     tier,
-				"toolchain.dev.openshift.com/owner":    username,
-				"toolchain.dev.openshift.com/type":     typeName,
-				"toolchain.dev.openshift.com/provider": "codeready-toolchain",
-			},
+			Name:   fmt.Sprintf("%s-%s", username, typeName),
+			Labels: labels,
 		},
 		Status: corev1.NamespaceStatus{Phase: corev1.NamespaceActive},
 	}
 	for _, set := range options {
-		set(ns)
+		set(ns, tier, typeName)
 	}
 	return ns
 }
 
-type namespaceOption func(*corev1.Namespace)
+type namespaceOption func(ns *corev1.Namespace, tier string, typeName string)
 
-func withRevision(revision string) namespaceOption { // nolint: unparam
-	return func(ns *corev1.Namespace) {
-		ns.ObjectMeta.Labels["toolchain.dev.openshift.com/revision"] = revision
+func withTemplateRef(revision string) namespaceOption { // nolint: unparam
```

Comment:
I believe the name of this func is a bit confusing. I was reviewing the test and commenting on all occurrences to say that the value was incorrect, until I reached here 🤦 

May I suggest that you rename this func, so we know that the argument is a revision and not a full templateRef value? Although, the result is an added `toolchain.dev.openshift.com/templateref` label on the namespace... 🤔 

So what about `withTemplateRevision(string)` instead?

Reply:
but the function doesn't add template `/revision` label to the namespace, but `/templateRef` label - then the name `withTemplateRevision` doesn't make sense

Reply:
> but the function doesn't add template /revision label to the namespace, but /templateRef label - then the name withTemplateRevision doesn't make sense

ok, but having `withTemplateRef("123456a")` looks confusing IMO. And I suggested `withTemplateRevision` and not `withRevision` because I know that it's not going to be a `/revision` label which is added on the namespace.

so maybe simplifying the func and pass the actual `/templateref` value as-is?

Reply:
I found pretty easily solution - the revision that is passed is always the same `abcde11` so I can easily remove the parameter and just keep `withTemplateRef()`
changed in 9ebe6f5

Reply:
that's simpler, indeed!